nft-market/pages/mint.tsx

```diff
@@ -0,0 +1,143 @@
+import { CalimeroSdk, WalletConnection } from "calimero-sdk";
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+import translations from "../constants/en.global.json";
+import { config } from "../utils/calimeroSdk";
+
+let walletConnectionObject: WalletConnection | undefined = undefined;
+
+interface MintProps {
+  walletConnection: WalletConnection | undefined;
+}
+
+interface NftForm {
+  title: string;
+  description: string;
+  media: string;
+}
+
+export default function Mint({ walletConnection }: MintProps) {
+  const router = useRouter();
+  const [isSignedIn, setIsSignedIn] = useState(false);
+  const [accountId, setAccountId] = useState<string | null>("");
+  const [mintVisible, setMintVisible] = useState(false);
+  const [nftsVisible, setNftsVisible] = useState(false);
+  const [nftsForSellVisible, setNftsForSellVisible] = useState(false);
+
+  const login = async () => {
+    try {
+      await walletConnectionObject?.requestSignIn({
+        contractId: config.nftContract,
+      });
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const logout = () => {
+    walletConnectionObject?.signOut();
+    setIsSignedIn(false);
+  };
+
+  useEffect(() => {
+    if (isSignedIn && localStorage.getItem("accountId")) {
+      setAccountId(localStorage.getItem("accountId"));
+    }
+  }, []);
+  const [nftForm, setNftForm] = useState<NftForm>({
+    title: "",
+    description: "",
+    media: "",
+  });
+
+  useEffect(() => {
+    const init = async () => {
+      const calimero = await CalimeroSdk.init(config).connect();
+      walletConnectionObject = new WalletConnection(
+        calimero,
+        config.nftContract
+      );
+      const signedIn = await walletConnectionObject?.isSignedInAsync();
+      const account = walletConnectionObject?.account();
+      if (account && signedIn) {
+        localStorage.setItem("accountId", account.accountId);
+      }
+      setIsSignedIn(signedIn);
+    };
+    init();
+  }, []);
+
+  useEffect(() => {
+    const absolute = window.location.href.split("?");
+    const url = absolute[0];
+    router.replace(url);
+  }, []);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+    setNftForm({ ...nftForm, [name]: value });
+  };
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    console.log(walletConnectionObject);
+
+    event.preventDefault();
+    const { title, description, media } = nftForm;
+    await mintAssetToNft(walletConnectionObject, title, description, media);
+  };
+
+  return (
+    <PageWrapper
+      isSignedIn={isSignedIn}
+      title={translations.home.title}
+      nearLogin={login}
+      nearLogout={() => {
+        logout();
+        router.reload();
+      }}
+      nearSignedIn={isSignedIn}
+      openMint={true}
+    >
+      <div className="flex justify-center text-white">
```

Comment:
this whole section needs to be extracted to component

Reply:
Made a mintDialog component